File: go/TwoSum/twosum.go
package twosum

func twoSum(nums []int, target int) []int {
	var sol []int
	cenas := map[int]int{}

	for i, num := range nums {
		nums2 := target - num
		if v, ok := cenas[nums2]; ok {
			sol = append(sol, i, v)
		} else {
			cenas[num] = i
		}
	}
	return sol
}

File: java/TwoSum/TwoSum.java
package java.TwoSum;

import java.util.HashMap;

public class TwoSum {

    public int[] Solution(int[] nums, int target) {
        int[] sol = {};

        HashMap<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int nums2 = target - nums[i];

            if (map.containsKey(nums2)) {

                sol[0] = i;
                sol[1] = map.get(nums2);
                return new int[] { i, map.get(nums2) };
            } else {
                map.put(nums[i], i);
            }

        }
        return sol;
    }
}